design/cdk-bootstrap.md

```diff
@@ -0,0 +1,476 @@
+# `cdk bootstrap`
+
+`cdk bootstrap` is a tool in the AWS CDK command-line interface responsible for populating a given environment
+(that is, a combination of AWS account and region)
+with resources required by the CDK to perform deployments into that environment.
+
+This document is a design for extending the capabilities of the `bootstrap` command motivated by the needs of the
+["Continuous delivery for CDK apps"](https://github.com/aws/aws-cdk/pull/3437) epic.
+
+## Required changes
+
+### Bootstrap resources
+
+The `bootstrap` command creates a CloudFormation stack in the environment passed on the command line.
+Currently, the only resource in that stack is:
+
+* An S3 bucket that holds the file assets and the resulting CloudFormation template to deploy.
+
+We will add the following resources to the bootstrap stack:
+
+* An ECR repository that stores the images that are the results of building Docker assets.
+
+If the `--trust` option has been passed,
```

Comment:
I am wondering, since we plan to use these roles even if we are not doing cross-environment deployments, they should be created anyway, no?

Reply:
I think that won't work though, right? If `--trust` has not been passed, then the roles will not have any principals in their trust policy, which is not allowed (nobody will be able to assume them).

Reply:
Can we always trust the current account?

Reply:
Done.